modules/apps/transfer/keeper/relay.go

```diff
@@ -277,6 +278,10 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 		return errorsmod.Wrap(err, "failed to mint IBC tokens")
 	}
 
+	if !k.bankKeeper.HasDenomMetaData(ctx, voucherDenom) {
```

Comment:
if we are running migrations such that all traces have metadata set, shouldn't we just make setting metadata a functionality of `SetDenomTrace`? This avoids accidental incidents where the trace is set but the metadata isn't. There should never be an instance when you want to set the denom trace without setting the metadata as well?